### PR TITLE
Add a broader restorer mechanism for WMS 

### PR DESCRIPTION
**Pull request by pblottiere**
## Description

This PR adds a broader restorer mechanism for the WMS service (cf the discussion on https://github.com/qgis/QGIS/pull/36007).

This way we can easily add specific restorers without updating the renderer itself.

### CODE REVIEW

File: src/server/services/wms/qgswmsrestorer.h [hunk 1]
```diff
@@ -64,4 +65,32 @@ class QgsLayerRestorer
...
+      QgsWmsRestorer( const QgsWmsRenderContext &context );
+
+      /**
+       * Default destructor.
+       */
+      ~QgsWmsRestorer() = default;
+
+    private:
+
+      std::unique_ptr<QgsLayerRestorer> mLayerRestorer;
```
Comment (nyalldawson) [inline src/server/services/wms/qgswmsrestorer.h:92]: Does this need to be a unique ptr? It could just be a value instead.

Comment (pblottiere) [reply src/server/services/wms/qgswmsrestorer.h:92]: The `QgsLayerRestorer` constructor needs a list of layers.

Comment (nyalldawson) [reply src/server/services/wms/qgswmsrestorer.h:92]: Yes but if you could initialise it:

QgsWmsRestorer::QgsWmsRestorer( const QgsWmsRenderContext &context )
    : mLayerRestorer( context.layers() )
{}

Comment (pblottiere) [reply src/server/services/wms/qgswmsrestorer.h:92]: Oh yes of course, silly me :upside_down_face:.

Thanks Nyall, I'm going to do that.

**Pull request merged**